models/topic.go

```diff
@@ -70,6 +97,52 @@ func GetTopicByName(name string) (*Topic, error) {
 	return &topic, nil
 }
 
+// addTopicByNameToRepo adds a topic name to a repo and increments the topic count.
+// Returns topic after the addition
+func addTopicByNameToRepo(repoID int64, topicName string, e Engine) (*Topic, error) {
+	var topic Topic
+	if has, err := e.Where("name = ?", topicName).Get(&topic); err != nil {
+		return nil, err
+	} else if !has {
```

Comment:
Limit multiple if else if ...
```suggestion
	} 
	if !has {
```